### ISSUE TITLE
EditorModelEventManager does not emit SendingContentModel event on adding / editing block item to the BlockList dditor

### ISSUE THREAD

**Issue by mateusz-kotwica** [umbraco/Umbraco-CMS#10792]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

8.15.0

### Bug summary

In the previous versions (for example 8.10) every time when new (empty) block item was added to the BlockList a `SendingContentModel` event was emitetd by `EditorModelEventManager`. It was allowing to perfrom some custom actions, for example to set default values for block item.

After latest update I found out that this event is raising no more. After short investigation I realized the difference is: `GetEmptyByKeys` in v8.15 in comparision with `GetEmptyByKey` is that, it returns a list and I think following event attribute [](https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web/WebApi/Filters/OutgoingEditorModelEventAttribute.cs) is skiped because it is expecting `ObjectContent objectContent`, but in fact in the current implementationa collection is passed as a content.

### Steps to reproduce

Just try to subscribe to `EditorModelEventManager.SendingContentModel` (for example in custom `IComponent` component) and try to add new item to the BlockList. 

### Expected result / actual result

The `EditorModelEventManager.SendingContentModel `event should be raised.

---
_This item has been added to our backlog [AB#13244](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/13244)_

**Comment by nul800sebastiaan** [comment]
We'll need to have a look, can you please tell us a little more about what you're using `SendingContentModel` for? Maybe a code example to help us understand why it's important for you to get that event?

**Comment by mateusz-kotwica** [comment]
Thank you @nul800sebastiaan for suacha quick response. Basically I'm using SendingContentModel event to handle situation when new (empty) block item is added to the BlockList. I'm setting then some default values to the BlockItem's properties. 

A pseudocode sample: 

```
private void SetDefaultValues(HttpActionExecutedContext sender, EditorModelEventArgs<ContentItemDisplay> e)
{
    // check if the request is for empty content. Based on Uri. Looking for either "GetEmpty" or "GetEmptyByKey" path
    var isRequestForEmptyContent = IsRequestForEmptyContent(sender.Request.RequestUri.AbsoluteUri);
          
    // just get preconfigured static data from helper class
    var defaultValuePropertyNameCollection = _defaultValues.GetDefaultValuesPropertyAliases(e.Model.ContentTypeAlias);
          
    // get all properties from the content model
    var allDocumentProperties = e.Model.Variants.FirstOrDefault()?.Tabs.SelectMany(tab => tab.Properties).ToList();

    foreach (var propertyName in defaultValuePropertyNameCollection)
    {
        var property = allDocumentProperties.FirstOrDefault(o => o.Alias.Equals(propertyName));       
        
        // perform update ...    
    }
}
```

**Comment by elit0451** [comment]
Fixed by #10869
